### ISSUE TITLE
Allow passing a template to the Cover Block

### ISSUE THREAD

**Issue by igmoweb** [WordPress/gutenberg#21869]
**Is your feature request related to a problem? Please describe.**
The current Cover Block doesn't allow any customization of its inner blocks template. Passing them through props could add that personalization possible to developers.

**Describe the solution you'd like**
A new prop for `CoverEdit` component that would allow to override the default template. Something like this (in `public_html/wp-content/plugins/gutenberg/packages/block-library/src/cover/edit.js` file):

```js
...
function CoverEdit( {
	attributes,
	setAttributes,
	isSelected,
	className,
	noticeUI,
	overlayColor,
	setOverlayColor,
	toggleSelection,
	noticeOperations,
	innerBlocksTemplate = INNER_BLOCKS_DEFAULT_TEMPLATE,
} ) {
...
```

Where `INNER_BLOCKS_DEFAULT_TEMPLATE` is defined already in that same file.

And finally:

```js
...
<div className="wp-block-cover__inner-container">
      <InnerBlocks template={ innerBlocksTemplate } />
</div>
...
```

This simple solution would allow to use the `editor.BlockEdit` to change the default template, like in this example:

```js
wp.hooks.addFilter( 'editor.BlockEdit', 'my-plugin/with-cover-template', ( BlockEdit ) => {
  return ( props ) => {
    const newProps = {
      ...props,
      innerBlocksTemplate: [[
        'core/heading',
        {
          placeholder: __( 'Write title…' ),
        },
      ]],
    };
    return <BlockEdit { ...newProps } />;
  };
} );
```

**Describe alternatives you've considered**
I've been wondering if passing this as a prop would be appropriate but passing it as an attribute is worse. The downside of this approach is that using the cover inside another `<InnerBlock />` won't allow customizing the cover by passing a different template, only the main template will be respected. Using attributes would solve that but it would not be a good practice.

Another alternative would be the use of a filter but it has the same cons that the previous approach.

**Comment by karmatosed** [comment]
It does seem that others are indicating this would be useful. That said, some time has passed so what I think is a good idea would be to get this given a technical feedback on if other methods can be done on this and/or if it still is something is an issue.

**Comment by tellthemachines** [comment]
What I'm wondering is if the use case for this would be covered by patterns. What problem is solved by passing a template to Cover? Would be good to know, especially as time has passed so there may be other ways of accomplishing whatever it is.

**Comment by mtias** [comment]
Yeah, closing this for now, but feel free to reopen if there's a use case that's not possible to be achieved with the current tools.
